src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -132,19 +132,56 @@ class ArchiveExplorer extends React.Component {
      */
     handleClickFullPath = fullPath => this.setState({ fullPath });
 
+    /**
+     * Handle search input, update view state
+     *
+     * @param {string} query - raw query string in the search bar
+     * @return {void}
+     */
+    handleSearch = query => {
+        const trimmedQuery = query.trim();
+        const newState = {
+            searchQuery: query,
+        };
+
+        if (!query) {
```

Comment:
It seems like this can be simplified to what's below:

```
this.setState({
    searchQuery: query,
    view: query.trim() ? VIEW_SEARCH : VIEW_FOLDER
});
```

Reply:
> It seems like this can be simplified to what's below:
> 
> ```
> this.setState({
>     searchQuery: query,
>     view: query.trim() ? VIEW_SEARCH : VIEW_FOLDER
> });
> ```

It's not the same as the original logic, because if `query && !trimmedQuery`, we don't change view state in the original code. But your advice makes sense and can make the logic simpler. Will do.

Reply:
Cool. Setting primitive values on state only really matters when the values change. Setting a primitive value to its current value should basically be a no-op.